### ISSUE TITLE
Alacritty deletes scrollback history scrollback on resuming after being hidden on Windows for Maximized/Fullscreen modes

### ISSUE THREAD

**Issue by yanchith** [alacritty/alacritty#7074]
Alacritty deletes scrollback history on resuming after being hidden on Windows for Maximized/Fullscreen modes. Doesn't happen for Windowed mode. Happens on two of my Windows machines. Didn't happen with previous versions of alacritty (I used 0.9 and 0.10 or something).

Video:

[alacritty-repro.webm](https://github.com/alacritty/alacritty/assets/1099245/5f0398a5-4629-4c79-9468-7f2b949019d6)

Config (sorry for zip, GH doesn't like yml):

[alacritty.zip](https://github.com/alacritty/alacritty/files/12025799/alacritty.zip)

Event Log:

[alacritty.log](https://github.com/alacritty/alacritty/files/12025792/alacritty.log)

### System

OS: Windows
Version: `alacritty 0.12.2 (9d9982d)`

**Comment by kchibisov** [comment]
Could you try `0.12.1` or `0.12.0`, and maybe `0.11.0`?

**Comment by yanchith** [comment]
Okay, this happens as far back as `0.11.0`, I just didn't update for a long while. Doesn't happen `0.10.1`, though.

**Comment by chrisduerr** [comment]
@kchibisov Winit is resizing to 0x0 on Windows, right? This might be clearing the grid.

**Comment by kchibisov** [comment]
@chrisduerr but we ignore such resizes on both path?

**Comment by chrisduerr** [comment]
We should, yes.

@yanchith Can you give the logs of windowed vs fullscreen mode? You say that only fullscreen has the issue, right?

**Comment by yanchith** [comment]
Hey, so sorry for the confusion. I just tried again, and I can reproduce this on all of Windowed, Maximized and Fullscreen, so this is not just specific to fullscreen.

The reason I got confused is that I can't reproduce this every time. Sometimes I can hide the window, restore it, and it text stays there, but at other times it disappears shortly (a few frames?) after the window is restored.

Here's a short log: [alacritty.zip](https://github.com/alacritty/alacritty/files/12071696/alacritty.zip)

When I get home, I'll try my other machine too.

**Comment by kchibisov** [comment]
If you know how to fix sure, but I think what happens is that conpty resets in between somehow.

**Comment by DrMeosch** [comment]
Also have this issue (ver. 0.12.2 Windows with WSL)

**Comment by kchibisov** [comment]
Could you try using custom conpty.dll (from windows terminal or wezterm), just putting it next to alacritty should make it load it iirc? Maybe it's a ConPty bug that got fixed... 

But from what you describe I don't think we should do anything, because:

1) system tells us to resize and we obey;
2) We resize the conpty handler as part of the resize otherwise things could desync (the same logic in on other systems).

So, alacritty logic is sane here, though, we don't delaying the PTY resizing here until we do `process_renderer_update`, it could help, but it's racy. Though, I'm not sure how we can easily test it, due to a fact that just moving won't work, we need to prevent terminal interactions with old size and handle window sizes larger than what terminal is. It all just feels like we're trying to workaround broken ConPty, which is just how it works.

If ConPty decides that it really wants to delete everything than it's a conpty issue, because it's not just a pipe it could manage the grid content, resize, etc.

**Comment by yanchith** [comment]
> Could you try using custom conpty.dll (from windows terminal or wezterm), just putting it next to alacritty should make it load it iirc? Maybe it's a ConPty bug that got fixed...

Tried `conpty.dll` from wezterm, and it behaves the same. Should have loaded successfully, because logs say: `[0.264224400s] [INFO ] [alacritty_terminal] Using conpty.dll for pseudoconsole`

> If ConPty decides that it really wants to delete everything than it's a conpty issue

The thing is this didn't happen on `0.10.1`. I'll try diffing around what changed. On the surface the code looks very similar.

**Comment by kchibisov** [comment]
It could be just upstream winit library changed, but given that you say `0.10.1` and not `0.11.0` it could take time to figure out what exactly changed it.

**Comment by yanchith** [comment]
You are right. winit jumped a couple of versions. The old version of winit used to resize to 0x0 when minimizing the window, which alacritty likely ignored, and thus didn't call `on_resize` for conpty. New winit resizes to 160x28, dunno why, their Windows backend seems to just straightforwardly translate WM_SIZE.

When resizing manually on `0.10.1` with windowed mode, I reproduce the same conpty bahavior, so the underlying bug was already present.

I wouldn't want to introduce weird hacks and workarounds here, so maybe I'll just patch my local alacritty with a workaround, unless you see a solution that's good enough to be upstreamed?

**Comment by kchibisov** [comment]
You can try to patch winit and figure out what changed.

**Comment by Long0x0** [comment]
Maybe try `PSEUDOCONSOLE_RESIZE_QUIRK`? It's an undocumented flag in conpty https://github.com/microsoft/terminal/issues/16911 .

**Comment by brian6932** [comment]
Probably same underlying cause as https://github.com/alacritty/alacritty/issues/6515#issuecomment-1489688789.
